astropy/units/tests/test_format.py

```diff
@@ -804,3 +804,20 @@ def test_celsius_fits():
     assert u.Unit("deg C kg-1", format="fits") == u.C * u.deg / u.kg
     assert u.Unit("Celsius kg-1", format="fits") == u.deg_C / u.kg
     assert u.deg_C.to_string("fits") == "Celsius"
+
+
+@pytest.mark.parametrize(
+    "format_spec, string",
+    [
+        ("generic", "dB(1 / m)"),
+        ("unscaled", "dB(1 / m)"),
+        ("latex", r"$\mathrm{dB}$$\mathrm{\left( \mathrm{\frac{1}{m}} \right)}$"),
+        ("latex_inline", r"$\mathrm{dB}$$\mathrm{\left( \mathrm{m^{-1}} \right)}$"),
+        ("console", "    1\ndB( -)\n    m"),
```

Comment:
Always good to see test cases! Why is there a space after the `(`? I think that is actually a buglet in the regular code...

Reply:
I think this comes because for some reason mutline fractional units are rendered with a leading space:
```python
>>> print(f'{(u.m**-1):unicode}')
 1
 ─
 m
````

Reply:
Indeed.  Which seems rather weird; their should be a space only if the unit has a scale.

Reply:
The spaces are a bit buggy; I think they should be resolved elsewhere. Also f.e.
```python
>>> f"{(u.eV*u.s**2).decompose()}"             # space between scale and unit
'1.60218e-19 kg m2'
>>> f"{(u.eV*u.s**2).decompose():unicode}"     # no space between scale and unit
'1.6021766×10⁻¹⁹m² kg'
>>> f"{(1*u.eV*u.s**2).decompose()}"           # space between value and unit
'1.602176634e-19 kg m2'
>>> f"{(1 * u.eV*u.s**2).decompose():unicode}" # space between value and unit
'1.602176634e-19 m² kg'
```
i.e. *no* space between factor and unit, but between value and unit.

Reply:
Agreed, this is a separate issue!

Reply:
See #14409